compiler/noirc_evaluator/src/ssa/ssa_gen/context.rs

```diff
@@ -544,8 +571,11 @@ impl<'a> FunctionContext<'a> {
     /// This is operationally equivalent to extract_current_value_recursive, but splitting these
     /// into two separate functions avoids cloning the outermost `Values` returned by the recursive
     /// version, as it is only needed for recursion.
-    pub(super) fn extract_current_value(&mut self, lvalue: &ast::LValue) -> LValue {
-        match lvalue {
+    pub(super) fn extract_current_value(
```

Comment:
I don't see why this and the recursive version are returning a Result, it seems to me it is always a Ok.

Reply:
`index_lvalue` and `codegen_array_index` can fail in both these methods